ui-common/src/main/java/io/syndesis/qe/hooks/UITestsHooks.java

```diff
@@ -1,20 +1,52 @@
 package io.syndesis.qe.hooks;
 
+import io.syndesis.qe.CustomWebDriverProvider;
+import io.syndesis.qe.TestConfiguration;
+
 import org.openqa.selenium.OutputType;
 import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.logging.LogEntry;
+import org.openqa.selenium.logging.LogType;
 
 import com.codeborne.selenide.WebDriverRunner;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 import io.cucumber.java.AfterStep;
 import io.cucumber.java.Scenario;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class UITestsHooks {
+
     @AfterStep
     public void afterScreenshot(Scenario scenario) {
         if (scenario.isFailed() && WebDriverRunner.hasWebDriverStarted()) {
             byte[] screenshotAsBytes = ((TakesScreenshot) WebDriverRunner.getWebDriver()).getScreenshotAs(OutputType.BYTES);
             scenario.attach(screenshotAsBytes, "image/png", "Screenshot");
+            printBrowserLog();
         }
     }
 
+    private void printBrowserLog() {
+        StringBuilder result = new StringBuilder("Browser console log:").append("\n");
+        if ("firefox".equals(TestConfiguration.syndesisBrowser())) {
+            // workaround for https://github.com/SeleniumHQ/selenium/issues/7792
+            try {
+                for (String entry : Files.readAllLines(Paths.get(CustomWebDriverProvider.FIREFOX_LOG_PATH), StandardCharsets.UTF_8)) {
+                    result.append(entry).append("\n");
+                }
+            } catch (IOException e) {
+                log.warn("Firefox browser log file doesn't exist.");
+            }
+        } else {
+            for (LogEntry entry : WebDriverRunner.getWebDriver().manage().logs().get(LogType.BROWSER)) {
+                result.append(entry.toString()).append("\n");
+            }
+        }
+        log.warn(result.toString());
```

Comment:
`attach` to scenario instead of printing in the logs as nobody checks the log anyway :grin: